scenarioo-server/src/main/java/org/scenarioo/business/diffViewer/comparator/AbstractComparator.java

```diff
@@ -36,21 +36,24 @@
 	protected static final ConfigurationRepository configurationRepository = RepositoryLocator.INSTANCE
 		.getConfigurationRepository();
 
+	protected ComparatorParameter parameter;
 	protected ScenarioDocuReader docuReader;
 	protected DiffWriter diffWriter;
 	protected String baseBranchName;
 	protected String baseBuildName;
 	protected ComparisonConfiguration comparisonConfiguration;
 
-	public AbstractComparator(String baseBranchName, String baseBuildName, final ComparisonConfiguration comparisonConfiguration) {
+	public AbstractComparator(ComparatorParameter parameter) {
+		this.parameter = parameter;
+		this.baseBranchName = parameter.getBaseBranchName();
+		this.baseBuildName = parameter.getBaseBuildName();
+		this.comparisonConfiguration = parameter.getComparisonConfiguration();
```

Comment:
The main problem is that before were three parameters. And this three parameters are passed through five classes at least. After adding one more value had to modify 5 classes and several test classes. Now pass through a configuration class. So no need to change so many files if adding or removing a parameter.